src/gsy_e/models/myco_matcher/myco_external_matcher.py

```diff
@@ -84,19 +84,25 @@ def publish_orders(self, message):
             for market in area_data["markets"]:
                 self.area_markets_mapping.update(
                     {f"{area_uuid}-{market.time_slot_str}": market})
-                market_offers_bids_list_mapping[area_uuid] = self._get_orders(market, filters)
+                if area_uuid not in market_orders_list_mapping:
+                    market_orders_list_mapping[area_uuid] = {}
+                market_orders_list_mapping[area_uuid].update(
+                    self._get_orders(market, filters))
 
             if area_data.get("future_markets"):
                 # Future markets
                 market = area_data["future_markets"]
                 self.area_markets_mapping.update(
                     {f"{area_uuid}-{time_slot_str}": market
                      for time_slot_str in market.orders_per_slot().keys()})
-                market_offers_bids_list_mapping[area_uuid] = self._get_orders(market, filters)
+                if area_uuid not in market_orders_list_mapping:
+                    market_orders_list_mapping[area_uuid] = {}
+                market_orders_list_mapping[area_uuid].update(
+                    self._get_orders(market, filters))
         self.area_uuid_markets_mapping = {}
         # TODO: change the `bids_offers` key and the channel to `orders`
         response_data.update({
-            "bids_offers": market_offers_bids_list_mapping,
+            "bids_offers": market_orders_list_mapping,
```

Comment:
Nit (unrelated to this PR): it would be good to change the `bids_offers` key to `orders`, as we discussed a while ago.

Reply:
I agree, let me create the PRs for fix after merging this fix, however I'm a bit concerned about backward compatibility, but let's fix it now rather than later